testdata/p4_16_samples/psa-multicast-basic-2-bmv2.stf

```diff
@@ -17,14 +17,14 @@ mc_node_associate 2 4
 #input packet
 packet  2 000000000001 000000000000 ffff   deadbeef deadbeef deadbeef deadbeef
 # expect ports
-expect  6 000000000001 000000000000 ffff   00000006 00000101 00000003 deadbeef $
-expect  7 000000000001 000000000000 ffff   00000007 00000102 00000003 deadbeef $
-expect  8 000000000001 000000000000 ffff   00000008 00000101 00000003 deadbeef $
-expect  8 000000000001 000000000000 ffff   00000008 00000102 00000003 deadbeef $
+expect  6 000000000001 000000000000 ffff   00000006 00000101 00000003 00000000 $
```

Comment:
is the value of the class_of_service guaranteed to be 0?

Reply:
Short answer: According to the PSA specification and how this P4_16 program, yes it is guaranteed.

Longer answer: The test P4 program initializes class_of_service during ingress processing using this assignment statement: `ostd.class_of_service = (ClassOfService_t) ((ClassOfServiceUint_t) hdr.ethernet.srcAddr[0:0]);`, so only tests the two possible class_of_service values of 0 or 1 in this program.  The PSA specification requires that for unicast and multicast packets, class_of_service is copied from ingress to egress unmodified, and from there the test program copies egress class_of_service into the output packet.